### DISCUSSION TITLE
Is it possible to delete a device?

### DISCUSSION THREAD

**Discussion by wsw70** [custom-components/pyscript#189] [Q&A]
I would like to periodically clean up stale ~~entries~~ devices in HA (usually plugs that are unused). Is there a way to delete ~~an entity~~ a device in HA from within `pyscript`?

**Comment by craigbarratt** [comment]
`state.delete("ENTITY_NAME")` or `del ENTITY_NAME` should both work.

**Reply by wsw70** [reply]
Thanks a lot (I missed that in https://hacs-pyscript.readthedocs.io/en/latest/reference.html?highlight=state.delete#state-variable-functions) but I actually meant DEVICE and not entity, sorry. (I corrected the question)
